src/providers/proxy/proxy_id.c

```diff
@@ -1488,6 +1488,152 @@ static int get_initgr(TALLOC_CTX *mem_ctx,
+    if (!grp) {
+        ret = ENOMEM;
+        DEBUG(SSSDBG_CRIT_FAILURE, "talloc() failed\n");
+        goto done;
+    }
+
+    // group 0 is the primary group so it can be skipped
+    for (j = 1; j < num_gids; j++) {
+        do {
+            /* always zero out the grp structure */
+            memset(grp, 0, sizeof(struct group));
+            buffer = grow_group_buffer(tmp_ctx, &buffer, &buflen);
+            if (!buffer) {
+                ret = ENOMEM;
+                goto done;
+            }
+
+            status = ctx->ops.getgrgid_r(gids[j], grp, buffer, buflen, &ret);
+
+            ret = handle_getgr_result(status, grp, dom, &delete_group);
+        } while (ret == EAGAIN);
+
+        if (ret != EOK) {
+            DEBUG(SSSDBG_OP_FAILURE,
+                    "getgrgid failed [%d]: %s\n", ret, strerror(ret));
+            goto done;
+        }
+
+        network_groupname = sss_create_internal_fqname(tmp_ctx, grp->gr_name, dom->name);
+        if (network_groupname == NULL) {
+            DEBUG(SSSDBG_OP_FAILURE, "Failed to create fqdn '%s'\n",
+                  grp->gr_name);
+            ret = ENOMEM;
+            goto done;
+        }
+
+        if (strstr(db_groupname, network_groupname) != NULL) {
+            *group_found = true;
+            ret = EOK;
+            goto done;
+        }
+    }
+
+done:
+    return ret;
+}
+
+static int remove_group_members(struct proxy_id_ctx *ctx,
+                                struct sysdb_ctx *sysdb,
+                                struct sss_domain_info *dom,
+                                struct passwd *pwd,
+                                long int num_gids,
+                                gid_t *gids)
+{
+    TALLOC_CTX *tmp_ctx;
+    int ret = EOK;
+    int i = 0;
+    int num_values = 0;
+    char *groupname = NULL;
+    const char *username = NULL;
+    const char *attr = SYSDB_MEMBEROF;
+    const char *attrs[] = {attr, NULL};
+    struct ldb_result *res = NULL;
+    struct ldb_message_element *el = NULL;
+    bool group_found = false;
+
+    tmp_ctx = talloc_new(NULL);
+    if (!tmp_ctx) {
+        ret = ENOMEM;
+        DEBUG(SSSDBG_CRIT_FAILURE, "talloc_new() failed\n");
+        goto done;
+    }
+
+    username = sss_create_internal_fqname(tmp_ctx, pwd->pw_name, dom->name);
+    if (username == NULL) {
+        ret = ENOMEM;
+        goto done;
+    }
+
+    ret = sysdb_get_user_attr(tmp_ctx, dom, username, attrs, &res);
+    if (res->count > 0) {
+        el = ldb_msg_find_element(res->msgs[0], attr);
+        if (el) {
+            num_values = el->num_values;
+        }
+    }
+
+    for (i = 0; i < num_values; i++) {
+        group_found = false;
+
+        ret = sysdb_get_rdn(sysdb, tmp_ctx, (const char *) el->values[i].data, NULL, &groupname);
+        if (ret != EOK) {
+            DEBUG(SSSDBG_CRIT_FAILURE, "Could not get rdn from [%s]\n",
+                  (const char *) el->values[i].data);
+            goto done;
```

Comment:
Here and in all other `goto done`: wouldn't it make sense to continue with next group?